enhancements/cluster-capabilities-api.md

```diff
@@ -0,0 +1,328 @@
+communicate expectations to the operators managing the cluster. This
+enhancement describes an API for expressing the expected
+"capabilities" of a cluster, including aspects such as how hard
+operators should work to provide high-availability and whether
+operators should configure their operands in a way to minimize
+resource consumption for especially constrained environments.
+
+## Motivation
+
+We want to avoid having cluster profiles use different manifests to
+configure operators managed by the cluster-version-operator, and
+instead rely on in-cluster APIs. This approach will make the same
+information available to operators managed by either the
+cluster-version-operator or operator-lifecycle-manager.
+
+We do not want to expose the cluster profile name directly to most of
+the operators, because we do not want to have to update operators each
+time we add a new profile.
+
+We want the settings to be global to the cluster, rather than using
+operator-specific APIs, because that simplifies the integration with
+the installer and allows them to be applied consistently in all
+operators that need them.
+
+### Goals
+
+- Expose capability expectations to operators running in a cluster so
+  they can configure their operands appropriately.
+
+### Non-Goals
+
+- Expose cluster profile names to operators.
+- Provide a mutable API for changing the cluster capabilities (at
+  least for now).
+- Predict other capability settings that we might need to expose.
+
+## Proposal
+
+A new [informational
+resource](https://docs.openshift.com/container-platform/4.6/installing/install_config/customizations.html#informational-resources_customizations),
+`capabilities.config.openshift.io`, is created to hold details about
+the configured features of a cluster. It will include two fields to
+start, and may be expanded over time.
+
+The `highAvailabilityMode` field will use an enum to express the
+high-availability expectations. Possible values are `full` and
+`none`. The default is `full`, which represents the behavior operators
+have today in a "normal" cluster. The `none` mode will be used in
+single-node deployments (developer and production), and the operator
+should not configure the operand for highly-available
+operation. Systems such as the assisted installer, which may
+experience periods early in the lifetime of the cluster when full HA
+is not possible, can start with `none` and transition to `full` when
+the cluster is more fully online.
+
+The `resourceLimits` fields will use an enum to express the general
+expectations for how much capacity operators should give their
+operands. Valid values for the enum are `small`, `medium`, and
+`large`. The default is `large`, representing the behavior for
+operators today. The `medium` configuration would be used to lower the
+resource use in single-node production clusters. For example, the
+cluster-monitoring-operator might deploy prometheus with a reduced set
```

Comment:
This makes sense, but determining what we do not need, e.g. what is reduced might be harder to achieve. This also depends on other components in the cluster that expose metrics we scrape, so something like setting sample limits to each target/component might be a good start.

I think we would need more specific resource requirements to know what medium and small means here, in this KEP.

Reply:
It does seem to be the consensus that we should move the sizing discussion to another enhancement, so I've cut this part of the text out for now.